src/week2.hs

```diff
@@ -30,6 +30,17 @@ exactMatches (x:xs) (x':xs')
     |x == x' = 1 + (exactMatches xs xs')
     |otherwise = (exactMatches xs xs')
 
+-- or, using zipWith (Dmitry's suggestion)
+sumIfEqual :: Peg -> Peg -> Int
+sumIfEqual x y = if (x == y) then 1 else 0
+
+exactMatches1 :: Code -> Code -> Int
+exactMatches1 x y = sum(zipWith (sumIfEqual) x y)
+
+-- or, even more concisely (though I'm not sure if this is better)
+exactMatches2 :: Code -> Code -> Int
+exactMatches2 x y = sum(zipWith (\a b -> if (a == b) then 1 else 0) x y)
```

Comment:
one compromise solution is to use a `where` clause... I think in this case both approaches are equally valid